debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -335,7 +335,15 @@ public void initializeStorage() {
         try (AdminClient admin = AdminClient.create(this.producerConfig.asProperties())) {
             // Find default replication factor
             Config brokerConfig = getKafkaBrokerConfig(admin);
-            final short replicationFactor = Short.parseShort(brokerConfig.get(DEFAULT_TOPIC_REPLICATION_FACTOR_PROP_NAME).value());
+            final short replicationFactor;
+            // Ensure that the default replication factor property was returned by the Admin Client
+            if (brokerConfig.get(DEFAULT_TOPIC_REPLICATION_FACTOR_PROP_NAME).value() != null) {
```

Comment:
It might be better to read the value into a variable and then test and covert it instead of repeating `(brokerConfig.get(DEFAULT_TOPIC_REPLICATION_FACTOR_PROP_NAME)` call.

Reply:
Good suggestion, I refactored this slightly as you recommended.